src/v1.rs

```diff
@@ -209,11 +210,22 @@ impl Party {
         aggregate_nonce: &Point,
         tweak: &Scalar,
     ) -> SignatureShare {
-        let mut z = &self.nonce.d + &self.nonce.e * compute::binding(&self.id(), nonces, msg);
-        z += compute::challenge(&(self.group_key + tweak * G), aggregate_nonce, msg)
+        let mut r = &self.nonce.d + &self.nonce.e * compute::binding(&self.id(), nonces, msg);
+        if tweak != &Scalar::zero() && !aggregate_nonce.has_even_y() {
+            r = -r;
+        }
+
+        let tweaked_public_key = self.group_key + tweak * G;
+        let mut cx = compute::challenge(&tweaked_public_key, aggregate_nonce, msg)
             * &self.private_key
             * compute::lambda(self.id, signers);
 
+        if tweak != &Scalar::zero() && !tweaked_public_key.has_even_y() {
+            cx = -cx;
+        }
+
+        let z = r + cx;
```

Comment:
Ah is this the mathy magic that makes it work? :P

Reply:
Yuuup :P